graphh/CGH.py

```diff
@@ -48,42 +51,84 @@ class GraphHopper(object):
                                   'time': 67674},
     ...
     """
+
+
     url = "https://graphhopper.com/api/1/"
 
     def __init__(self, api_key, premium=False):
+
         self.api_key = api_key
         self.premium = premium
 
-    def _url_handle(self, api, l_parameters):
-        """This function does an url request with given parameters
+
+
+    def _url_requests(self, api, data, request="get"):
+
+        """ This function does an url request ( GET or POST) with given parameters
 
         Parameters
         ----------
-         api: str
+        api: str
             name of the api used
-         l_parameters: list
-            list of parameters to insert in the url
+        data: dict
+            dict of parameters to insert in the url
 
         Returns
         -------
         dict
-            The dictionary return by url request
-         """
+            The dictionary return by url request (GET or POST)
+        """
+
         complete_url = GraphHopper.url + api + "?"
-        for p in l_parameters:
-            complete_url += "&{}".format(p)
-        complete_url += "&key=" + self.api_key
+
+        if request.upper() == "POST":
+            complete_url += "key={}".format(self.api_key)
```

Comment:
Is there a reson why the key is not added to the data dict as for the GET query?